src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java

```diff
@@ -478,6 +472,17 @@ else if (left.getSymbol().getType() == DPLLexer.INDEX_IN) {
 
             }
         }
+        else if (left.getSymbol().getType() == DPLLexer.SOURCETYPE && operation.getSymbol().getType() == DPLLexer.IN) {
+            for (DPLParser.StringTypeContext stringType : ctx.stringType()) {
+                final String sourceType = new UnquotedText(new TextString(stringType.getText())).read();
+                final String rlikeStatement = glob2rlike(sourceType);
+                if (sQualifier == null) {
```

Comment:
Is null needed here? Working with Spark columns is quite difficult without null and the for loops kind of require the check. But could it be possible to initialize the first `sQualifier` outside the for loop so null is not needed? Or maybe new objects.

Reply:
added OrColumn object